src/storage/store/chunked_node_group.cpp

```diff
@@ -243,6 +243,16 @@ bool ChunkedNodeGroup::delete_(const Transaction* transaction, row_idx_t rowIdxI
     return versionInfo->delete_(transaction, rowIdxInChunk);
 }
 
+void ChunkedNodeGroup::addColumn(Transaction* transaction, TableAddColumnState& addColumnState,
+    bool enableCompression) {
+    auto numRows = getNumRows();
+    auto chunkData = 
```

Comment:
I think you can directly construct a ColumnChunk here, which will create the ColumnChunkData for you.